src/VisualStudio/Core/Def/Implementation/VisualStudioMetadataAsSourceFileSupportService.cs

```diff
@@ -16,20 +19,24 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
     [Export(typeof(VisualStudioMetadataAsSourceFileSupportService))]
     internal sealed class VisualStudioMetadataAsSourceFileSupportService : IVsSolutionEvents
     {
+        private readonly IThreadingContext _threadingContext;
         private readonly IMetadataAsSourceFileService _metadataAsSourceFileService;
 
-#pragma warning disable IDE0052 // Remove unread private members - Used to store the AdviseSolutionEvents cookie.
-        private readonly uint _eventCookie;
-#pragma warning restore IDE0052 // Remove unread private members
```

Comment:
cookie is never read.

Reply:
I prefer this be kept so it's clear we're tracking the registration cookie